client/src/components/mapping/FhirMappingPanel/FhirResourceTree/index.tsx

```diff
@@ -225,121 +193,77 @@ const FhirResourceTree = ({ onClickCallback }: Props) => {
-    } else if (node.isArray) {
+    attribute.spreadTypes().map(buildNodeFromAttribute);
+
+  const buildNodeFromAttribute = (
+    attribute: Attribute
+  ): ITreeNode<Attribute> => {
+    const node = createNode(attribute, []);
+
+    if (attribute.types.length > 1) {
+      // if the attribute has multiple types, we have to create as many
+      // children as there are types.
+      node.childNodes = buildMultiTypeNode(attribute);
+    } else if (attribute.isArray) {
       // If node is array, we need to replicate the root node for this type
       // childNode will be the node really having the structure defined by
       // the structure definition.
-      childNodes = buildChildNodesForArray(node, definition);
-    } else if (node.definition.$slice) {
-      // if the node is a slicing, generate the sliced children
-      childNodes = genSlicedChildren(node.definition.$slice, node);
-    } else if (node.definition.$children) {
+      node.childNodes = buildChildNodesForArray(node);
+    } else if (attribute.slices.length > 0) {
+      // if the node has slices, create a node for each of them
+      node.childNodes = genTreeLevel(attribute.slices);
+    } else if (attribute.children.length > 0) {
       // if the node has children we already know about, create them.
-      childNodes = genTreeLevel(node.definition.$children, node);
-    } else {
-      // otherwise the node has no children for now.
-      childNodes = [];
+      node.childNodes = genTreeLevel(attribute.children);
     }
 
-    return createNode(node, childNodes, () =>
-      setNodes(nodes => addNodeToArray(nodes, node, definition.$children))
-    );
+    return node;
   };
 
-  const genSlicedChildren = (slices: any, parent: Node): ITreeNode<Node>[] =>
-    Object.values(slices).map(sliceDef =>
-      buildNodeFromDefinition(sliceDef, parent)
-    );
+  const genTreeLevel = (attributes: Attribute[]): ITreeNode<Attribute>[] =>
+    attributes.map(buildNodeFromAttribute);
 
-  const genTreeLevel = (display: any, parent?: Node): ITreeNode<Node>[] => {
-    return (
-      Object.keys(display)
-        // we don't want to display the $meta attribute, filter it out
-        .filter(attributeName => attributeName !== '$meta')
-        .map(attributeName =>
-          buildNodeFromDefinition(display[attributeName], parent)
-        )
-    );
+  const fetchAttributeDefinition = async (
+    definitionId: string
+  ): Promise<Attribute[]> => {
+    const { data } = await client.query({
+      query: qStructureDisplay,
+      variables: { definitionId }
+    });
+    if (!data || !data.structureDefinition) return [];
+    return data.structureDefinition.attributes.map(Attribute.from);
   };
 
-  const handleNodeClick = async (node: ITreeNode<Node>): Promise<void> => {
-    if (node.nodeData?.isArray) {
+  const handleNodeClick = async (node: ITreeNode<Attribute>): Promise<void> => {
+    const attribute = node.nodeData!;
+    console.debug(attribute.path);
```

Comment:
Don't forget this

Reply:
does it bother you if I keep it for now ? I find it convenient to understand what's going on (even on staging it might be useful)